library/L1_Peripheral/lpc40xx/capture.hpp

```diff
@@ -0,0 +1,285 @@
+        sjsu::cortex::InterruptController();
+
+    explicit constexpr Capture(
+        const CaptureChannel_t &timer,
+        const sjsu::SystemController &system_controller =
+            DefaultSystemController(),
+        const sjsu::InterruptController &interrupt_controller =
+            kInterruptController
+        )
+        : timer_(timer),
+          system_controller_(system_controller),
+          interrupt_controller_(interrupt_controller)
+    {
+    };
+
+    Status Initialize(
+        units::frequency::hertz_t frequency,
+        IsrPointer isr = nullptr, int32_t priority = -1) const override
+    {
+        // NOTE: Same initialization as Timer library's Initialize()
+        system_controller_.PowerUpPeripheral(timer_.channel.power_id);
+        SJ2_ASSERT_FATAL(
+            frequency != 0_Hz,
+            "Cannot have zero ticks per microsecond, please choose 1 or more.");
+
+        // Set Prescale register for Prescale Counter to milliseconds
+        uint32_t prescaler =
+            system_controller_
+                .GetPeripheralFrequency(timer_.channel.power_id) / frequency;
+        timer_.channel.timer_register->PR = prescaler;
+
+        // Enable timer
+        constexpr bit::Mask kTimerEnableBit = bit::CreateMaskFromRange(0);
+        timer_.channel.timer_register->TCR = bit::Set(
+                timer_.channel.timer_register->TCR, kTimerEnableBit.position);
+
+        // Install capture ISR
+        *timer_.channel.user_callback = isr;
+        interrupt_controller_.Register({
+                .interrupt_request_number = timer_.channel.irq,
+                .interrupt_service_routine = timer_.isr,
+                .enable_interrupt = true,
+                .priority = priority
+        });
+
+        // Disable both capture channels
+        ConfigureCapture(kCaptureChannel0, kCaptureEdgeModeNone);
+        ConfigureCapture(kCaptureChannel1, kCaptureEdgeModeNone);
+        EnableCaptureInterrupt(kCaptureChannel0, false);
+        EnableCaptureInterrupt(kCaptureChannel1, false);
+
+        return Status::kSuccess;
+    }
+
+    void GetCaptureStatus(CaptureChannelNumber channel, CaptureStatus_t &status)
+    {
+        status.flags = timer_.channel.timer_register->IR;
+
+        if (channel == kCaptureChannel0)
+        {
+          status.count = timer_.channel.timer_register->CR0;
+        }
+        else
+        {
+          status.count = timer_.channel.timer_register->CR1;
+        }
+    }
+
+    void ConfigureCapture(
+            CaptureChannelNumber channel,
+            CaptureEdgeMode mode) const override
+    {
+        if (channel == kCaptureChannel1)
+        {
+            timer_.channel.capture_pin1.SetPinFunction(3);
+        }
+        else
+        {
+            timer_.channel.capture_pin0.SetPinFunction(3);
+        }
+
+        static constexpr bit::Mask kModeBits[2] =
+        {
+            bit::CreateMaskFromRange(0, 1),
+            bit::CreateMaskFromRange(3, 4)
+        };
```

Comment:
Something is also doable with bit::Masks is that they are actually just a structure with a uint8_t for position and width. Rather than making an array of these you could simply do this:

```suggestion
        static constexpr bit::Mask kModeBits =
        {
            .position =  3 * which,
            .width = 2,
        };
```
This will eliminate some of the code below. Be sure to comment why this is being done though.